rules/trackhub.smk

```diff
@@ -96,38 +99,169 @@ rule narrowpeak_bignarrowpeak:
+            direction2=forward
+        fi
+                    
+        bamCoverage --bam {input.bam} --outFileName {output.forward} --filterRNAstrand $direction1 --numberOfProcessors {threads} {params.flags} --verbose >> {log} 2>&1 &&        
+        bamCoverage --bam {input.bam} --outFileName {output.reverse} --filterRNAstrand $direction2 --numberOfProcessors {threads} {params.flags} --verbose >> {log} 2>&1
+        """
+
+rule bam_bigwig:
+    """
+    Convert a bam file into a bigwig file
+    """
+    input:
+        bam=expand("{dedup_dir}/{{assembly}}-{{sample}}.{{sorter}}-{{sorting}}.bam", **config),
+        bai=expand("{dedup_dir}/{{assembly}}-{{sample}}.{{sorter}}-{{sorting}}.bam.bai", **config)
+    output:
+        expand("{result_dir}/bigwigs/{{assembly}}-{{sample}}.{{sorter}}-{{sorting}}.bw", **config)
+    params:
+        config['bam_bigwig']['deeptools'] if config.get('bam_bigwig', False) else ""
+    wildcard_constraints:
+        sorting=config['bam_sort_order'] if config.get('bam_sort_order') else ""
+    log:
+        expand("{log_dir}/bam_bigwig/{{assembly}}-{{sample}}.{{sorter}}-{{sorting}}.log", **config),
+    benchmark:
+        expand("{benchmark_dir}/bam_bigwig/{{assembly}}-{{sample}}.{{sorter}}-{{sorting}}.benchmark.txt", **config)[0]
+    conda:
+        "../envs/deeptools.yaml"
+    threads: 20
+    resources:
+        deeptools_limit=1
+    shell:
+        """
+        bamCoverage --bam {input.bam} --outFileName {output} --numberOfProcessors {threads} {params} --verbose >> {log} 2>&1
+        """
+
+
+rule twobit:
+    """
+    Generate a 2bit file for each assembly
+    """
+    input:
+        expand("{genome_dir}/{{assembly}}/{{assembly}}.fa", **config)
+    output:
+        expand("{genome_dir}/{{assembly}}/{{assembly}}.2bit", **config)
+    log:
+        expand("{log_dir}/assembly_hub/{{assembly}}.2bit.log", **config)
+    benchmark:
+        expand("{benchmark_dir}/assembly_hub/{{assembly}}.2bit.benchmark.txt", **config)[0]
+    conda:
+        "../envs/ucsc.yaml"
+    shell:
+        "faToTwoBit {input} {output}"
+
+
+def get_trackhub_files(wildcards):
+    """
+
+    """
+    trackfiles = {key: [] for key in ['bigwigs', 'bigpeaks', 'twobits', 'annotations']}
+
+    # check whether or not each assembly is supported by ucsc
+    for assembly in set(samples['assembly']):
+        # check for response of ucsc
+        response = requests.get(f"https://genome.ucsc.edu/cgi-bin/hgTracks?db={assembly}",
+                                allow_redirects=True)
+        assert response.ok, "Make sure you are connected to the internet"
+
+        # see if the title of the page mentions our assembly
+        if not any(assembly == x for x in
+                   re.search(r'<TITLE>(.*?)</TITLE>', response.text).group(1).split(' ')):
+            trackfiles['twobits'].append(f"{config['genome_dir']}/{assembly}/{assembly}.2bit")
+
+        # now check if we have annotation for
+        # TODO
```

Comment:
for later?

Reply:
Yes, its waiting for the annotation downloads and a bugfix from Snakemake (which they actually fixed last week :D). 

That part is necessary for nice assembly hub with genes, but not essential for assembly hub functionality